R/derive_summary_records.R

```diff
@@ -196,12 +196,39 @@ derive_summary_records <- function(dataset,
     subset_ds <- dataset
   }
 
+  # Summarise the analysis value and bind to the original dataset
   # Summarise the analysis value and bind to the original dataset
   bind_rows(
     dataset,
-    subset_ds %>%
-      group_by(!!!by_vars) %>%
-      summarise(!!analysis_var := summary_fun(!!analysis_var)) %>%
-      mutate(!!!set_values_to)
+    get_summary_records(
+      dataset,
+      by_vars = by_vars,
+      filter = !!filter,
+      analysis_var = !!analysis_var,
+      summary_fun = summary_fun,
+      set_values_to = set_values_to)
   )
+  get_summary_records <- function(dataset,
```

Comment:
The `get_summary_records()` function should not be defined *within* the `derive_summary_records()` function but as a separate function. Otherwise, you can not call it in `derive_params_exposure()`.

Please add a header to document the `get_summary_records()` function. It could be useful for users. Thus I would export it. For the `@keywords` tag I would use `adam user_utility`.

Reply:
updated the code. Please check.

Reply:
The `get_summary_records()` function is still defined within the `derive_summary_records()` function.

Reply:
separated the code from the main code.